KlarnaOnDemand/KODLocalization.m

```diff
@@ -1,17 +1,28 @@
 #import "KODLocalization.h"
+#import "KODContext.h"
 
 @implementation KODLocalization
 
 + (NSBundle*)KODBundle {
   static NSBundle *bundle = nil;
+  static NSString *locale = nil;
   
-  if (!bundle) {
+  if (!bundle || locale != [KODContext getPreferredLocale]) {
+    locale = [KODContext getPreferredLocale];
     NSString *bundlePath = [NSBundle.mainBundle pathForResource:@"KOD" ofType:@"bundle"];
     if (!bundlePath) {
       bundlePath = [[NSBundle bundleForClass:[self class]] pathForResource:@"KOD" ofType:@"bundle"];
     }
     bundle = [NSBundle bundleWithPath:bundlePath];
+    
+    if (locale != [[NSBundle mainBundle] preferredLocalizations].firstObject) {
```

Comment:
to make the code more readable, you can extract `[[NSBundle mainBundle] preferredLocalizations]` to a method `getDeviceLocale` and reuse it here.